html/code/modules/modules/xarinit.php

```diff
@@ -20,47 +20,41 @@
  */
 function modules_init()
 {
-    // Get database information
+    // Create tables inside a transaction
     $dbconn = xarDB::getConn();
-    $tables =& xarDB::getTables();
-
     $prefix = xarDB::getPrefix();
+    try {
+        $dbconn->begin();
+
+        sys::import('xaraya.tableddl');
+        xarXMLInstaller::createTable('table_schema-def', 'modules');
 
-    $tables['modules'] = $prefix . '_modules';
-    $tables['module_vars'] = $prefix . '_module_vars';
-    $tables['module_itemvars'] = $prefix . '_module_itemvars';
-    $tables['hooks'] = $prefix . '_hooks';
+        // We're done, commit
+        $dbconn->commit();
+    } catch (Exception $e) {
+        $dbconn->rollback();
+        throw $e;
+    }
+    // Get database information
+   // $dbconn = xarDB::getConn();
+    $tables =& xarDB::getTables();
     $tables['eventsystem'] = $prefix . '_eventsystem';
+
+    //$prefix = xarDB::getPrefix();
+
+    //$tables['modules'] = $prefix . '_modules';
+    //$tables['module_vars'] = $prefix . '_module_vars';
+    //$tables['module_itemvars'] = $prefix . '_module_itemvars';
+    //$tables['hooks'] = $prefix . '_hooks';
+    
     
     // Create tables
     // This should either go, or fail competely
-    try {
-        $charset = xarSystemVars::get(sys::CONFIG, 'DB.Charset');
-        $dbconn->begin();
-        /**
-         * Here we create all the tables for the module system
-         *
-         * prefix_modules       - basic module info
-         * prefix_module_vars   - module variables table
-         * prefix_hooks         - table for hooks
-         */
+    //try {
+        //$charset = xarSystemVars::get(sys::CONFIG, 'DB.Charset');
+        //$dbconn->begin();
         // prefix_modules
-        /**
-         * CREATE TABLE xar_modules (
-         *   id int(11) NOT NULL auto_increment,
-         *   name varchar(64) NOT NULL,
-         *   regid int(10) integer unsigned NOT NULL,
-         *   directory varchar(64) NOT NULL,
-         *   version varchar(10) NOT NULL default '0',
-         *   class varchar(64) NOT NULL,
-         *   category varchar(64) NOT NULL,
-         *   admin_capable INTEGER NOT NULL default '0',
-         *   user_capable INTEGER NOT NULL default '0',
-         *   state INTEGER NOT NULL default '0'
-         *   PRIMARY KEY  (id)
-         * )
-         */
-        $fields = array(
+       /* $fields = array(
```

Comment:
No need to comment out. Just remove the unneeded code

Reply:
okay we will remove.